mesh/nurbs.cpp

```diff
@@ -45,6 +45,45 @@ KnotVector::KnotVector(int order, int NCP)
    knot = -1.;
 }
 
+KnotVector::KnotVector(int order, const Vector& intervals,
+                       const Array<int>& continuity )
+{
+   MFEM_ASSERT(continuity.Size() == (intervals.Size() + 1),
+               "Incompatible sizes of continuity and intervals.");
+   Order = order;
+   NumOfElements = intervals.Size();
+   const int num_knots = Order * continuity.Size() - continuity.Sum();
+   NumOfControlPoints = num_knots - Order - 1;
+   knot.SetSize(num_knots);
+   real_t accum = 0.0;
+   int iknot = 0;
+   for (int i = 0; i < intervals.Size(); ++i)
+   {
+      const int multiplicity = max(1, Order - continuity[i]);
+      for (int j = 0; j < multiplicity; ++j)
+      {
+         knot[iknot] = accum;
+         ++iknot;
+      }
+      accum += intervals[i];
+   }
+   if (intervals.Size() < continuity.Size())
```

Comment:
This is always true, but I suppose this check is in anticipation of implementing the periodic case later?

Reply:
I simplified the `KnotVector` constructor in the latest commit and explicitly mention in the documentation string now that periodicity is not supported.